typed-racket-lib/typed-racket/types/subtype.rkt

```diff
@@ -502,33 +510,58 @@
         [else #f])]
      [(Evt: evt-t)
       (cond
-        [(memq kind '(Semaphore
-                      Output-Port
-                      Input-Port
-                      TCP-Listener
-                      Thread
-                      Subprocess
-                      Will-Executor))
+        [(member t1 (list -Semaphore
```

Comment:
Hmm... maybe, but I think its clearer to keep the current structure.

On a related note, I just moved all of the static data declarations in this file (such as this list) to the module level so they are only allocated once.